### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by EPashkin**
cc @GuillaumeGomez, @sdroege

### CODE REVIEW

**Comment by EPashkin** [comment]
Last only 3 warnings, I don't fixed now as it breaking change, but IMHO this better changed in future.
`pub struct FontFace(Shared<ffi::cairo_font_face_t>);`
`pub struct ScaledFont(Shared<ffi::cairo_scaled_font_t>);`
```
warning: this argument is passed by value, but not consumed in the function body
  --> src\font\scaled_font.rs:42:27
   |
42 |     pub fn new(font_face: FontFace, font_matrix: &Matrix, ctm: &Matrix, options: &FontOptions) -> ScaledFont {
   |                           ^^^^^^^^ help: consider taking a reference instead: `&FontFace`
   |
   = note: #[warn(needless_pass_by_value)] on by default
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.197/index.html#needless_pass_by_value

warning: this argument is passed by value, but not consumed in the function body
   --> src\context.rs:665:44
    |
665 |     pub fn set_font_face(&self, font_face: FontFace) {
    |                                            ^^^^^^^^ help: consider taking a reference instead: `&FontFace`
    |
    = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.197/index.html#needless_pass_by_value

warning: this argument is passed by value, but not consumed in the function body
   --> src\context.rs:677:48
    |
677 |     pub fn set_scaled_font(&self, scaled_font: ScaledFont) {
    |                                                ^^^^^^^^^^ help: consider taking a reference instead: `&ScaledFont`
    |
    = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.197/index.html#needless_pass_by_value
```

File: src/context.rs [hunk 1]
```diff
@@ -622,7 +622,8 @@ impl Context {
 
     pub fn select_font_face(&self, family: &str, slant: FontSlant, weight: FontWeight) {
         unsafe {
-            ffi::cairo_select_font_face(self.0, CString::new(family).unwrap().as_ptr(), slant, weight)
+            let family = CString::new(family).unwrap();
+            ffi::cairo_select_font_face(self.0, family.as_ptr(), slant, weight)
```
Comment (sdroege) [inline src/context.rs]: Nice catch, these are actually causing memory unsafety in practice

**Comment by sdroege** [comment]
Looks good to me

File: src/context.rs (hunk 1 above)
Comment (EPashkin) [reply src/context.rs]: All thanks for clippy

**Comment by GuillaumeGomez** [comment]
Looks good to me as well. Thanks!

**Pull request merged**